browser/ui/views/side_panel/brave_side_panel.h

```diff
@@ -44,21 +53,30 @@ class BraveSidePanel : public views::View,
   // views::ResizeAreaDelegate:
   void OnResize(int resize_amount, bool done_resizing) override;
 
+  // views::View:
+  void ChildVisibilityChanged(View* child) override;
+  void OnThemeChanged() override;
+  gfx::Size GetMinimumSize() const override;
+  void AddedToWidget() override;
+
   void SetMinimumSidePanelContentsWidthForTesting(int width) {}
 
  private:
+  friend class sidebar::SidebarBrowserTest;
+
   void UpdateVisibility();
   void UpdateBorder();
-
-  // views::View:
-  void ChildVisibilityChanged(View* child) override;
-  void OnThemeChanged() override;
+  void OnSidebarWidthChanged();
 
   // views::ViewObserver:
   void OnChildViewAdded(View* observed_view, View* child) override;
   void OnChildViewRemoved(View* observed_view, View* child) override;
 
-  HorizontalAlignment horizontal_alighment_ = kHorizontalAlignLeft;
+  HorizontalAlignment horizontal_alignment_ = kHorizontalAlignLeft;
+  int starting_width_on_resize_ = -1;
```

Comment:
optional: I think having this `optional<int>` could be more symantic.

Reply:
Done.